webgpu.h

```diff
@@ -21,1002 +21,1004 @@
-    wgpuProcBufferReference bufferReference;
-    wgpuProcBufferRelease bufferRelease;
-
-    wgpuProcCommandBufferReference commandBufferReference;
-    wgpuProcCommandBufferRelease commandBufferRelease;
-
-    wgpuProcCommandEncoderFinish commandEncoderFinish;
-    wgpuProcCommandEncoderBeginComputePass commandEncoderBeginComputePass;
-    wgpuProcCommandEncoderBeginRenderPass commandEncoderBeginRenderPass;
-    wgpuProcCommandEncoderCopyBufferToBuffer commandEncoderCopyBufferToBuffer;
-    wgpuProcCommandEncoderCopyBufferToTexture commandEncoderCopyBufferToTexture;
-    wgpuProcCommandEncoderCopyTextureToBuffer commandEncoderCopyTextureToBuffer;
-    wgpuProcCommandEncoderCopyTextureToTexture commandEncoderCopyTextureToTexture;
-    wgpuProcCommandEncoderInsertDebugMarker commandEncoderInsertDebugMarker;
-    wgpuProcCommandEncoderPopDebugGroup commandEncoderPopDebugGroup;
-    wgpuProcCommandEncoderPushDebugGroup commandEncoderPushDebugGroup;
-    wgpuProcCommandEncoderReference commandEncoderReference;
-    wgpuProcCommandEncoderRelease commandEncoderRelease;
-
-    wgpuProcComputePassEncoderInsertDebugMarker computePassEncoderInsertDebugMarker;
-    wgpuProcComputePassEncoderPopDebugGroup computePassEncoderPopDebugGroup;
-    wgpuProcComputePassEncoderPushDebugGroup computePassEncoderPushDebugGroup;
-    wgpuProcComputePassEncoderSetPipeline computePassEncoderSetPipeline;
-    wgpuProcComputePassEncoderSetBindGroup computePassEncoderSetBindGroup;
-    wgpuProcComputePassEncoderDispatch computePassEncoderDispatch;
-    wgpuProcComputePassEncoderDispatchIndirect computePassEncoderDispatchIndirect;
-    wgpuProcComputePassEncoderEndPass computePassEncoderEndPass;
-    wgpuProcComputePassEncoderReference computePassEncoderReference;
-    wgpuProcComputePassEncoderRelease computePassEncoderRelease;
-
-    wgpuProcComputePipelineReference computePipelineReference;
-    wgpuProcComputePipelineRelease computePipelineRelease;
-
-    wgpuProcDeviceCreateBindGroup deviceCreateBindGroup;
-    wgpuProcDeviceCreateBindGroupLayout deviceCreateBindGroupLayout;
-    wgpuProcDeviceCreateBuffer deviceCreateBuffer;
-    wgpuProcDeviceCreateBufferMapped deviceCreateBufferMapped;
-    wgpuProcDeviceCreateCommandEncoder deviceCreateCommandEncoder;
-    wgpuProcDeviceCreateComputePipeline deviceCreateComputePipeline;
-    wgpuProcDeviceCreateRenderPipeline deviceCreateRenderPipeline;
-    wgpuProcDeviceCreatePipelineLayout deviceCreatePipelineLayout;
-    wgpuProcDeviceCreateQueue deviceCreateQueue;
-    wgpuProcDeviceCreateSampler deviceCreateSampler;
-    wgpuProcDeviceCreateShaderModule deviceCreateShaderModule;
-    wgpuProcDeviceCreateTexture deviceCreateTexture;
-    wgpuProcDeviceCreateBufferMappedAsync deviceCreateBufferMappedAsync;
-    wgpuProcDeviceReference deviceReference;
-    wgpuProcDeviceRelease deviceRelease;
-
-    wgpuProcFenceGetCompletedValue fenceGetCompletedValue;
-    wgpuProcFenceOnCompletion fenceOnCompletion;
-    wgpuProcFenceRefe
```

Comment:
can we move all the refcounting stuff into a separate header, even if it's hosted alongside webgpu.h?

Reply:
Unified them instead to not be per device but on `WGPUObject` which is `void*`